File: Windows/GameControllerSimulator/BluetoothLibrary/IntConverter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BluetoothLibrary
{
    public static class IntConverter
    {
        public static byte[] ToBigEndian(int value)
        {
            return new byte[]
            {
            (byte)(value >> 24),
            (byte)(value >> 16),
            (byte)(value >> 8),
            (byte)value
            };
        }

        public static byte[] ToLittleEndian(int value)
        {
            return new byte[]
            {
            (byte)value,
            (byte)(value >> 8),
            (byte)(value >> 16),
            (byte)(value >> 24)
            };
        }

        public static int FromBigEndian(byte[] bytes)
        {
            if (bytes == null || bytes.Length < 4)
                throw new ArgumentException("ByteArray size must be at least 4");

            return
                (bytes[0] << 24) |
                (bytes[1] << 16) |
                (bytes[2] << 8) |
                bytes[3];
        }

        public static int FromLittleEndian(byte[] bytes)
        {
            if (bytes == null || bytes.Length < 4)
                throw new ArgumentException("ByteArray size must be at least 4");

            return
                bytes[0] |
                (bytes[1] << 8) |
                (bytes[2] << 16) |
                (bytes[3] << 24);
        }
    }
}

File: iOS/GameControllerSimulator/CommonKit/IntConverter.swift
//
//  IntConverter.swift
//  GameControllerSimulator
//
//  Created by 猫眼螺 on 2026/7/29.
//

import Foundation

public enum IntConverter {

    /// Int 转大端字节序
    public static func toBigEndian(_ value: Int) -> [UInt8] {
        return [
            UInt8((value >> 24) & 0xFF),
            UInt8((value >> 16) & 0xFF),
            UInt8((value >> 8) & 0xFF),
            UInt8(value & 0xFF)
        ]
    }


    /// Int 转小端字节序
    public static func toLittleEndian(_ value: Int) -> [UInt8] {
        return [
            UInt8(value & 0xFF),
            UInt8((value >> 8) & 0xFF),
            UInt8((value >> 16) & 0xFF),
            UInt8((value >> 24) & 0xFF)
        ]
    }


    /// 大端字节序转 Int
    public static func fromBigEndian(_ bytes: [UInt8]) -> Int {
        precondition(bytes.count >= 4, "Byte array size must be at least 4")

        return (Int(bytes[0]) << 24) |
               (Int(bytes[1]) << 16) |
               (Int(bytes[2]) << 8) |
                Int(bytes[3])
    }


    /// 小端字节序转 Int
    public static func fromLittleEndian(_ bytes: [UInt8]) -> Int {
        precondition(bytes.count >= 4, "Byte array size must be at least 4")

        return Int(bytes[0]) |
               (Int(bytes[1]) << 8) |
               (Int(bytes[2]) << 16) |
               (Int(bytes[3]) << 24)
    }
}
